warehouse/query-core/src/main/java/datawave/query/jexl/functions/GroupingRequiredFilterFunctions.java

```diff
@@ -93,9 +92,12 @@
                         if (!rightSideMatches.isEmpty()) {
                             matches.addAll(rightSideMatches);
                             matches.add(match); // add the left side unmodified match
+                            tailHasMatch = true;
```

Comment:
This whole block within the ValueTuple is duplicated in the Iterable section.  Consider refactoring it out into a separate helper method.